libs/feature/search/src/lib/fuzzy-search/fuzzy-search.component.ts

```diff
@@ -72,7 +72,9 @@ export class FuzzySearchComponent implements OnInit {
     if (this.inputSubmitted.observers.length > 0) {
       this.inputSubmitted.emit(any)
     } else {
-      this.searchService.updateFilters({ any })
+      this.searchFacade.searchId === 'myRecords'
+        ? this.searchService.setFilters({ any })
+        : this.searchService.updateFilters({ any })
```

Comment:
This is a very specific change on a pretty generic component. Maybe you could listen to `inputSubmitted` in the `SearchHeaderComponent` and remove the owner filter from there if you're on myRecords (detecting this via a `ViewChild` or checking the current URL path for example) ?